common/include/pcl/common/impl/transforms.hpp

```diff
@@ -306,6 +305,52 @@ transformPointCloud (const pcl::PointCloud<PointT> &cloud_in,
 }
 
 
+void
+transformPointCloud(pcl::PointCloud<pcl::PointXY>& cloud_in, 
+                    pcl::PointCloud<pcl::PointXY>& cloud_out, 
+                    const Eigen::Affine2f& transform, 
+                    bool copy_all_fields)
+  {
+    if (&cloud_in != &cloud_out)
+    {
+      cloud_out.header   = cloud_in.header;
+      cloud_out.is_dense = cloud_in.is_dense;
+      cloud_out.reserve (cloud_in.size ());
+      if (copy_all_fields)
+        cloud_out.assign (cloud_in.begin (), cloud_in.end (), cloud_in.width);
+      else
+        cloud_out.resize (cloud_in.width, cloud_in.height);
+      cloud_out.sensor_orientation_ = cloud_in.sensor_orientation_;
+      cloud_out.sensor_origin_      = cloud_in.sensor_origin_;
+    }
+    if(cloud_in.is_dense)
+    {
+      for (std::size_t i = 0; i < cloud_out.size (); ++i)
+      {
+        Eigen::Vector2f pf = cloud_in[i].getVector2fMap();
```

Comment:
Have you tried `cloud_out[i].getVector2fMap() = transform * cloud_in[i].getVector2fMap()`?

Reply:
It gives this error.
```
error: invalid use of non-static member function pcl::Vector2fMap pcl::PointXY::getVector2fMap() 
```

Reply:
That's a weird one. Some more error context please

Reply:
I mean that's the first error. the rest I received are followups.